containers/podman-desktop
# fix: Check for Kubernetes connection before retrieving pods

fix: Check for Kubernetes connection before retrieving pods

### What does this PR do?

* Gets rid of the ECONNREFUSED connection errors when first starting
  Podman Desktop with a Podman Machine running
* Adds a check (listing namespaces) for an active connection before
  running `kubernetesListPods`.
* Moves podsInfos.set(result) to higher up as sometimes we are not
  checking Kubernetes if it's not active

### Screenshot/screencast of this PR



N/A

### What issues does this PR fix or reference?



PR created while investigating: https://github.com/containers/podman-desktop/issues/2096

### How to test this PR?



Start Podman Desktop with developer console and a podman machine running with kind STOPPED. You should no longer see
any ECONNREFUSED when first starting Podman Desktop.

Signed-off-by: Charlie Drage <charlie@charliedrage.com>
